### DISCUSSION TITLE
32-bit and 64-bit branch in the same project?

### DISCUSSION THREAD

**Discussion by joshbooker** [joyfullservice/msaccess-vcs-addin#411] [General]
Anybody have experience with a 32-bit and 64-bit branch in the same project? 

I mean it's probably not ideal and I intend to get the servers on same bitness eventually but are there pitfalls using VCS? Besides handling the 'PtrSafe' declarations are there other pitfalls like code reference guids or something? 

If so, thanks in advance for saving me the hassle as I won't even try it.

TIA

**Comment by bclothier** [comment]
I think a clarification is needed here. Are you trying to build both 32-bit ACCDE and 64-bit ACCDE? If that is your goal, then you need to have 2 machines, one with 32-bit Office installed, other with 64-bit Office installed and build it on each machine. You cannot have both 32-bit adn 64-bit ACCDEs on the same machine. 

When working with ACCDB, as long you have `LongPtr` and VBA7 declarations, it should run regardless of the bitness. 

This implies that for add-in, you can only run only whatever the bitness the Office is. It should automatically select the correct version of add-in.

**Comment by joshbooker** [comment]
Thanks for your reply.  I'm not talking ade build in Access, talking VCS build.  

After some testing, I found the 64-bit declarations indeed compile on 32bit Access.  My misunderstanding, only having experience going one-way (from 32 to 64) was that these changes were no longer able to compile in 32-bit.  I stand corrected.  Learn something new everyday.

So there is no need to maintain two branches for bitness.  Also, the VBA references appear the same on both so not a problem building from the same VCS source in either Access 32bit or Access 64bit.
